extensions/amp-fx-flying-carpet/0.1/amp-fx-flying-carpet.js

```diff
@@ -102,7 +101,13 @@ export class AmpFlyingCarpet extends AMP.BaseElement {
     });
     if (this.firstLayoutCompleted_) {
       this.scheduleLayout(this.children_);
-      listen(this.element, AmpEvents.BUILT, this.layoutBuiltChild_.bind(this));
+      this.children_.forEach(child => {
```

Comment:
This isn't equivalent. We're listening for new children to be added to the flying carpet, not for already know children to upgrade. "Necessary since not all children will be built by the time the flying-carpet has its #layoutCallback called."

Let's split this into a separate PR.

Reply:
Added possibly more complete impl. to #23767